bundles/org.openhab.binding.cbus/src/main/java/org/openhab/binding/cbus/handler/CBusNetworkHandler.java

```diff
@@ -0,0 +1,290 @@
+            }
+            String state = network.getState();
+            logger.debug("Network state is {}", state);
+            if ("new".equals(state)) {
+                projectObject.start();
+                logger.debug("Need to wait for it to be synced");
+            } else if ("sync".equals(state)) {
+                logger.debug("Network is syncing so wait for it to be ok");
+            }
+            if (!"ok".equals(state)) {
+                ScheduledFuture<?> initNetwork = this.initNetwork;
+                if (initNetwork == null || initNetwork.isCancelled()) {
+                    this.initNetwork = scheduler.scheduleWithFixedDelay(this::checkNetworkOnline, 30, 30,
+                            TimeUnit.SECONDS);
+                    logger.debug("Schedule a check every minute");
+                } else
+                    logger.debug("initNetwork alreadys started");
+                updateStatus();
+                return;
+            }
+            logger.debug("State should be ok !!!!!!");
+        } catch (CGateException e) {
+            logger.warn("Cannot load C-Bus network {}", networkID, e);
+            updateStatus(ThingStatus.UNINITIALIZED, ThingStatusDetail.COMMUNICATION_ERROR);
+        }
+        updateStatus();
+    }
+
+    private void checkNetworkOnline() {
+        Network network = getNetwork();
+        try {
+            if (network != null && network.isOnline()) {
+                logger.debug("Network is online");
+                ScheduledFuture<?> initNetwork = this.initNetwork;
+                if (initNetwork != null) {
+                    initNetwork.cancel(false);
+                }
+                updateStatus();
+            } else {
+                ThingStatus lastStatus = getThing().getStatus();
+                logger.debug("Network still not online {}", lastStatus);
+                updateStatus();
+            }
+        } catch (CGateException e) {
+            logger.warn("Cannot check if network is online {} ", network.getNetworkID());
+            updateStatus();
+        }
+    }
+
+    private void updateStatus() {
+        logger.debug("updateStatus");
+        ThingStatus lastStatus = getThing().getStatus();
+        Network network = getNetwork();
+        CBusCGateHandler cbusCGateHandler = getCBusCGateHandler();
+        try {
+            if (cbusCGateHandler == null || !cbusCGateHandler.getThing().getStatus().equals(ThingStatus.ONLINE)) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE, "CGate connection offline");
+            } else if (network == null) {
+                logger.debug("No network - set configuration error");
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR);
+            } else if (network.isOnline()) {
+                logger.debug("Network is online");
+                updateStatus(ThingStatus.ONLINE);
+            } else {
+                logger.debug("Network is offline");
```

Comment:
I don't see a purpose in this logging statement, the thing status change is already logged for you. Same for most of the other logs in the methods as well.

Reply:
Simplified the logging here.